src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/AbstractCloudFoundryDeployer.java

```diff
@@ -76,5 +83,27 @@ String buildpack(AppDeploymentRequest request) {
 	Predicate<Throwable> isNotFoundError() {
 		return t -> t instanceof AbstractCloudFoundryException && ((AbstractCloudFoundryException) t).getStatusCode() == HttpStatus.NOT_FOUND.value();
 	}
+	/**
+	 * To be used in order to retry the status operation for an application or task.
+	 * @param id The application id or the task id
+	 * @param <T> The type of status object being queried for, usually AppStatus or TaskStatus
+	 * @return The function that executes the retry logic around for determining App or Task Status
+	 */
+	<T> Function<Mono<T>, Mono<T>> statusRetry(String id) {
+		long requestTimeout = Math.round(this.deploymentProperties.getStatusTimeout() * 0.20); // wait 200ms with status timeout of 1000ms
+		long initialRetryDelay = Math.round(this.deploymentProperties.getStatusTimeout() * 0.10); // wait 100ms with status timeout of 1000ms
+
+		return m -> m.timeout(Duration.ofMillis(requestTimeout))
+			.doOnError(e -> logger.error(String.format("Error getting status for %s within %sms, Retrying operation.", id, requestTimeout), e))
```

Comment:
The exception thrown here is `TimeoutException` and the stack trace doesn't add value so will remove the logging of the exception, also for the other doOnError below. 

```12:47:33.999 reactor-http-nio-2 cloudfoundry-client.request       GET    https://api.run.pivotal.io:443/v2/spaces/2d4eeb03-e1f3-4c29-8f10-84bd144680f2/apps?q=name%20IN%20testSimpleDeployment-0a60d142-c2bb-4b6f-95c5-2abbb903fdb1&page=1
12:47:34.783 timer-1            org.springframework.cloud.deployer.spi.cloudfoundry.ErrorHandlingUtils Error getting status for testSimpleDeployment-0a60d142-c2bb-4b6f-95c5-2abbb903fdb1 within 2000, Retrying operation.
java.util.concurrent.TimeoutException: null
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.handleTimeout(FluxTimeout.java:243)
	at reactor.core.publisher.FluxTimeout$TimeoutMainSubscriber.doTimeout(FluxTimeout.java:227)
	at reactor.core.publisher.FluxTimeout$TimeoutTimeoutSubscriber.onNext(FluxTimeout.java:338)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.FluxResume$ResumeSubscriber.onNext(FluxResume.java:71)
	at reactor.core.publisher.FluxOnAssembly$OnAssemblySubscriber.onNext(FluxOnAssembly.java:316)
	at reactor.core.publisher.MonoDelay$MonoDelayRunnable.run(MonoDelay.java:96)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Assembly trace from producer [reactor.core.publisher.MonoTimeout] :
	reactor.core.publisher.Mono.timeoutMillis(Mono.java:2822)
	reactor.core.publisher.Mono.timeoutMillis(Mono.java:2783)
	reactor.core.publisher.Mono.timeoutMillis(Mono.java:2768)
	reactor.core.publisher.Mono.timeout(Mono.java:2700)
	org.springframework.cloud.deployer.spi.cloudfoundry.ErrorHandlingUtils.lambda$statusRetry$4(ErrorHandlingUtils.java:45)
	reactor.core.publisher.Mono.transform(Mono.java:2885)
	org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryAppDeployer.getStatus(CloudFoundryAppDeployer.java:271)
	org.springframework.cloud.deployer.spi.cloudfoundry.CloudFoundryAppDeployer.deploy(CloudFoundryAppDeployer.java:100)
	org.springframework.cloud.deployer.spi.test.AbstractAppDeployerIntegrationTests.testSimpleDeployment(AbstractAppDeployerIntegrationTests.java:96)
	org.springframework.test.context.junit4.statements.RunBeforeTestMethodCallbacks.evaluate(RunBeforeTestMethodCallbacks.java:75)
	org.springframework.test.context.junit4.statements.RunAfterTestMethodCallbacks.evaluate(RunAfterTestMethodCallbacks.java:86)
	org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	org.junit.rules.RunRules.evaluate(RunRules.java:20)
	org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:84)
	org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:252)
	org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:94)
	org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)
	org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:70)
	org.springframework.cloud.deployer.spi.test.junit.AbstractExternalResourceTestSupport$1.evaluate(AbstractExternalResourceTestSupport.java:73)
	org.junit.rules.RunRules.evaluate(RunRules.java:20)
	org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:191)
Composition chain until failing Operator :
	|_	Mono.timeout(ErrorHandlingUtils.java:45)
```